src/lib/mongoDriver/mongoConnectionPool.cpp

```diff
@@ -329,54 +330,81 @@ static std::string composeMongoUri
 {
   // Compose the mongoUri, taking into account all information
 
-  std::string uri = "mongodb://";
+  std::string uri;
 
-  // Add auth parameter if included
-  if (strlen(username) != 0 && strlen(passwd) != 0)
+  if (strlen(dbURI) != 0)
   {
-    uri += username + std::string(":") + passwd + "@";
-  }
+    if (strlen(username) != 0 || strlen(authDb) != 0 || strlen(rplSet) != 0 || strlen(mechanism) != 0 || dbSSL || dbDisableRetryWrites || timeout > 0)
+    {
+        LM_X(1, ("Invalid Command Line Options: -dbURI cannot be combined with -dbhost, -rplSet, -dbTimeout, -dbuser, -dbAuthMech, -dbAuthDb, -dbSSL and -dbDisableRetryWrites"));
+    }
 
-  uri += host + std::string("/");
+    const char* pwd = strstr(dbURI, "${PWD}");
+    if (pwd != NULL)
+    {
+      if (strlen(passwd) == 0)
+      {
+        LM_X(1, ("Invalid Command Line Options: -dbURI is used with a password substitution, but no password (-dbpwd) is supplied"));
+      }
 
-  if (strlen(authDb) != 0)
-  {
-    uri += authDb;
+      uri = std::string(dbURI, pwd - dbURI) + passwd + (pwd + 6);
```

Comment:
```suggestion
      // +6 is the length of the "${PWD}"
      uri = std::string(dbURI, pwd - dbURI) + passwd + (pwd + 6);
```

To explain from where that number comes